### PR TITLE
[asl] Update stdlib definition of `Replicate`

### PR DESCRIPTION
**Pull request by hrutvik**

### CODE REVIEW

File: asllib/libdir/stdlib.asl [hunk 1]
```diff
@@ -119,15 +119,15 @@ end;
...
-  assert (N MOD M == 0);
-  if N == 1 then
+  if M == 1 then
     return ReplicateBit{N}(IsZero(x));
   else
-    var r: bits(N) = Zeros{}();
-    for i=0 to (N DIV M)-1 do
-      r[i*:M] = x;
+    let items = N DIV M; // must be exact
+    var result : bits(N) = Zeros{}();
```
[the remark this answers is not in the document]
Comment (hrutvik) [reply asllib/libdir/stdlib.asl:126]: @nigelito do you have a preference?

File: asllib/libdir/stdlib.asl [hunk 2]
```diff
@@ -119,15 +119,15 @@ end;
 
 func Replicate{N,M}(x: bits(M)) => bits(N)
 begin
-  assert (N MOD M == 0);
-  if N == 1 then
+  if M == 1 then
```
Comment (Roman-Manevich) [inline asllib/libdir/stdlib.asl:122]: What is the purpose of this? If the purpose of the code is to serve as an example, there's no need for optimizations.

Comment (maranget) [reply asllib/libdir/stdlib.asl:122]: This code is called in mode `-variant asl` and the optimisation had some impact. I do not know whether this is still the  case or not.

File: asllib/libdir/stdlib.asl (hunk 1 above)
[the remark this answers is not in the document]
Comment (nigelito) [reply asllib/libdir/stdlib.asl:126]: The proposed change looks goof to me.

File: asllib/libdir/stdlib.asl (hunk 2 above)
Comment (nigelito) [reply asllib/libdir/stdlib.asl:122]: My preference would be not to include this optimisation in example/reference code, but I can live with it.

Comment (Roman-Manevich) [reply asllib/libdir/stdlib.asl:122]: We can opt to include this optimization in an internally-used version of the standard library.

**Comment by maranget** [comment]
FYI, I have just checked that restoring the optimisation (it was in fact disabled) yields a speedup of about four on test `herd/tests/instructions/AArch64.ASL/SXTW01.litmus`.

**Pull request merged**